taipy/core/_scheduler/_scheduler.py

```diff
@@ -195,3 +204,38 @@ def _update_job_config(cls):
             cls._dispatcher = _StandaloneJobDispatcher()
         else:
             cls._dispatcher = _DevelopmentJobDispatcher()
+
+    @classmethod
+    def _set_process_in_scheduler(cls, job_id, process):
+        cls._processes[job_id] = process
+
+    @classmethod
+    def _pop_process_in_scheduler(cls, job_id, default=None):
+        return cls._processes.pop(job_id, default)  # type: ignore
+
+    @classmethod
+    def _cancel_job(cls, job):
+        if process := cls._pop_process_in_scheduler(job.id):
+            process.cancel()  # TODO: this doesn't cancel the running process
+        job.cancelled()
+        cls.__unlock_edit_on_outputs(job)
+
+        # if job in cls._needs_to_run:
+        #     cls._needs_to_run.remove(job)
+
+        if job in cls.blocked_jobs:
+            cls.blocked_jobs.remove(job)
```

Comment:
Needs to be locked first.

Reply:
we're only looking at cls.blocked_jobs so the job is already locked